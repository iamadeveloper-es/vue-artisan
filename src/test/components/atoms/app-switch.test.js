// sum.test.js
import { shallowMount } from '@vue/test-utils';
import VaSwitch from '../../../components/atoms/va-switch/index.vue';
import { expect, test, describe, beforeEach } from 'vitest';



const wrapper = shallowMount(VaSwitch, {
  props: {}
});

describe('va-money', () => {

  beforeEach(async () => {
    // Clear mocks and add some testing data after before each test run
    await wrapper.setProps({
      amount: 3525.05
    });
  });

  test('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  //   describe('Modelos', () => {

  //     test('Seteo de amount a 1475.56, currency y lang por defecto es = a 1.475,56€', async () => {
  //       const props = {
  //         amount: 1475.56
  //       };
  //       await wrapper.setProps(props);

  //       const formatedAmountNoSpaces = wrapper.vm.formatedNumber.replace(/\s/g, '');

  //       expect(wrapper.vm.formatedNumber).not.toBeUndefined();
  //       expect(formatedAmountNoSpaces).toBe('1.475,56€');
  //       expect(formatedAmountNoSpaces).not.toBe('€1.475,56');

//     });
//   });
});
