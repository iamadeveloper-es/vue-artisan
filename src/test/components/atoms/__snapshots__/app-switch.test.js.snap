// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app-money > snapshot 1`] = `
<div
  amount="3525.05"
  class="app-switch-wrapper"
>
  <label
    class="app-switch"
    for="app-switch-1"
  >
    <input
      id="app-switch-1"
      role="checkbox"
      type="checkbox"
      value=""
    />
    <span
      class="app-switch__slider"
    />
  </label>
</div>
`;
